Public/Src/Tools/Execution.Analyzer/Analyzers.Partner/ContentPlacementAnalyzer.cs

```diff
@@ -0,0 +1,532 @@
+            Dictionary<uint, int> scheduledPipDependencyCount,
+            Dictionary<uint, int> scheduledPipInputCount,
+            Dictionary<uint, int> scheduledPipOutputCount,
+            PathTable pathTable,
+            StringTable stringTable)
+        {
+            var output = new List<BXLPipData>();
+            foreach (var eData in executionData)
+            {
+                output.Add(BuildFromExecutionData(pip, eData, scheduledPipDependencyCount, scheduledPipInputCount, scheduledPipOutputCount, pathTable, stringTable));
+            }
+            return output;
+        }
+
+        public static BXLPipData BuildFromExecutionData(Pip pip, PipExecutionPerformanceEventData executionData,
+            Dictionary<uint, int> scheduledPipDependencyCount,
+            Dictionary<uint, int> scheduledPipInputCount,
+            Dictionary<uint, int> scheduledPipOutputCount,
+            PathTable pathTable,
+            StringTable stringTable)
+        {
+            var pipData = new BXLPipData();
+            pipData.SemiStableHash = pip.FormattedSemiStableHash;
+            pipData.DurationMs = (executionData.ExecutionPerformance.ExecutionStop - executionData.ExecutionPerformance.ExecutionStart).TotalMilliseconds;
+            pipData.Type = pip.PipType.ToString();
+            pipData.ExecutionLevel = executionData.ExecutionPerformance.ExecutionLevel.ToString();
+            pipData.StartTimeTicks = executionData.ExecutionPerformance.ExecutionStart.Ticks;
+            pipData.TagCount = pip.Tags.Count();
+            int depCount;
+            int iCount;
+            int oCount;
+            scheduledPipDependencyCount.TryGetValue(pip.PipId.Value, out depCount);
+            scheduledPipInputCount.TryGetValue(pip.PipId.Value, out iCount);
+            scheduledPipOutputCount.TryGetValue(pip.PipId.Value, out oCount);
+            // and set these ones too
+            pipData.DependencyCount = depCount;
+            pipData.InputCount = iCount;
+            pipData.OutputCount = oCount;
+            if (pip.PipType == PipType.Process)
+            {
+                var process = pip as Pips.Operations.Process;
+                pipData.Priority = process.Priority;
+                pipData.Weight = process.Weight;
+                pipData.SemaphoreCount = process.Semaphores.Length;
+            }
+            // done...
+            return pipData;
+        }
+
+    }
+
+    public class BXLArtifact
+    {
+        public string Hash { get; set; } = null;
+        public string ReportedFile { get; set; } = null;
+        public long ReportedSize { get; set; } = -1;
+        public string BuidId { get; set; } = null;
+        public string BuildQueue { get; set; } = null;
+        public long BuildStartTimeTicks { get; set; } = 0;
+        public double BuildDurationMs { get; set; } = 0;
+        public int TotalPips { get; set; } = 0;
+        public int NumInputPips { get; set; } = 0;

```

Comment:
>ips { get; set; } = 0; [](start = 28, length = 22)

Have you considered making this a computed property?
It seems this is manually couingit the items in InputPuts.

You can write:
`public int NumInputPips => InputPips.Count`
so you don't have to do the book keeping yourself.